### ISSUE TITLE
Consuming nested state files for different environments without overwriting default values

### ISSUE THREAD

**Issue by janekbaraniewski** [roboll/helmfile#873]
Hey! :)
I have an issue with nested states and I'm not sure if that's because I misunderstood some concepts or is it just some missing functionality.
##### use case
I want to be able to consume predefined helmfiles with releases of e.g. databases, message brokers, monitoring tools etc. so I can quickly build different environments depending on my needs. I want to be able to overwrite default values supplied with those releases but I also want to be able to just use defaults. I want to be able to apply this configuration to different environments.
##### solution
My solution with nested states looks similar to this:

```yaml
# environment.yaml
helmfiles:
  - path: helmfile_grafana.yaml
  - path: helmfile_application.yaml
    values:
      - some_app_values.yaml
```

```yaml
# helmfile_grafana.yaml
repositories:
- name: stable
  url: https://kubernetes-charts.storage.googleapis.com

helmDefaults:
  tillerNamespace: example 
  tillerless: true
  kubeContext: kube-context          

releases:
  - name: grafana
    namespace: default
    chart: stable/grafana
    version: 3.8.12 
    values:
      - default_values.yaml
```
##### problem 

This works great on `default` environment but when I change my `environment.yaml` to
```yaml
# environment.yaml
helmfiles:
  - path: helmfile_grafana.yaml
  - path: helmfile_application.yaml
    values:
      - some_app_values.yaml

environments:
  env1:
    values:
      - env_values.yaml
```
and run helmfile with `--environment env1` my Grafana release is just skipped. 
When I add some dummy value to grafana nested state:
```yaml
# environment.yaml
helmfiles:
  - path: helmfile_grafana.yaml
    values:
      - dummy: value
  - path: helmfile_application.yaml
    values:
      - some_app_values.yaml

environments:
  env1:
    values:
      - env_values.yaml
```
Everything starts working fine. 

So at first I thought - this must be bug. But then I started debugging and I came to `create.go:209`
```golang
	} else if ctxEnv == nil && name != DefaultEnv {
		return nil, &UndefinedEnvError{msg: fmt.Sprintf("environment \"%s\" is not defined", name)}
	}
```
and I realized that maybe I'm just using environments in a completely wrong way - since I only define them at the top helmfile level and  (what I realized only after debugging) they are not available in nested state scope. Only after I specify some dummy value a `ctxEnv` exists and is later merged into nested state environments. It seems unintuitive to me that values for specific nested state are part of `ctxEnv` but values specified at the top level for the whole environment in  `env_values.yaml` are not but maybe my intuition just failed me.  

So, my question is - is this by design? If so - do you have some ideas on how I could achieve my goal without having to include some dummy value to every nested state file? And if not - is this a functionality that could be supported by helmfile?
